baseband/mark4/base.py

```diff
@@ -389,14 +389,15 @@ def read(self, count=None, fill_value=0., out=None):
 
         return out
 
-    def _read_frame(self):
+    def _read_frame(self, fill_value=0.):
         frame_nr = self.offset // self.samples_per_frame
         self.fh_raw.seek(self.offset0 + frame_nr * self.header0.framesize)
         self._frame = self.read_frame(ntrack=self.header0.ntrack,
                                       decade=self.header0.decade)
         # Convert payloads to data array.
         self._frame_data = self._frame.data
         self._frame_nr = frame_nr
+        self._frame.invalid_data_value = fill_value
```

Comment:
I think you need to move this up above the `.data` call, since that does the actual decoding and invalid data setting.